e2e/cases.go

```diff
@@ -8,8 +8,89 @@ import (
+				ID: to.Ptr(defaultImageVersionIDs["ubuntu2204"]),
+			}
+		},
+	},
+	"marinerv1": {
+		bootstrapConfigMutator: func(t *testing.T, nbc *datamodel.NodeBootstrappingConfiguration) {
+			nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-cblmariner-v1"
+			nbc.AgentPoolProfile.Distro = "aks-cblmariner-v1"
+		},
+		vmConfigMutator: func(vmss *armcompute.VirtualMachineScaleSet) {
+			vmss.Properties.VirtualMachineProfile.StorageProfile.ImageReference = &armcompute.ImageReference{
+				ID: to.Ptr(defaultImageVersionIDs["marinerv1"]),
+			}
+		},
+	},
+	"marinerv2": {
+		bootstrapConfigMutator: func(t *testing.T, nbc *datamodel.NodeBootstrappingConfiguration) {
+			nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-cblmariner-v2-gen2"
+			nbc.AgentPoolProfile.Distro = "aks-cblmariner-v2-gen2"
+		},
+		vmConfigMutator: func(vmss *armcompute.VirtualMachineScaleSet) {
+			vmss.Properties.VirtualMachineProfile.StorageProfile.ImageReference = &armcompute.ImageReference{
+				ID: to.Ptr(defaultImageVersionIDs["marinerv2"]),
+			}
+		},
+	},
+	"ubuntu2204-arm64": {
+		bootstrapConfigMutator: func(t *testing.T, nbc *datamodel.NodeBootstrappingConfiguration) {
+			nbc.ContainerService.Properties.AgentPoolProfiles[0].VMSize = "Standard_D2pds_V5"
+			nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-ubuntu-arm64-containerd-22.04-gen2"
+			// This needs to be set based on current CSE implementation...
+			nbc.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.CustomKubeBinaryURL = "https://acs-mirror.azureedge.net/kubernetes/v1.24.9/binaries/kubernetes-node-linux-arm64.tar.gz"
+			nbc.K8sComponents.PodInfraContainerImageURL = "mcr.microsoft.com/oss/kubernetes/pause:3.6"
+			nbc.KubeletConfig["--pod-infra-container-image"] = "mcr.microsoft.com/oss/kubernetes/pause:3.6"
+			nbc.AgentPoolProfile.VMSize = "Standard_D2pds_V5"
+			nbc.AgentPoolProfile.Distro = "aks-ubuntu-arm64-containerd-22.04-gen2"
+			nbc.IsARM64 = true
+
+		},
+		vmConfigMutator: func(vmss *armcompute.VirtualMachineScaleSet) {
+			vmss.Properties.VirtualMachineProfile.StorageProfile.ImageReference = &armcompute.ImageReference{
+				ID: to.Ptr(defaultImageVersionIDs["ubuntu2204-arm64"]),
+			}
+			vmss.SKU.Name = to.Ptr("Standard_D2pds_V5")
+		},
+	},
+	"marinerv2-arm64": {
+		bootstrapConfigMutator: func(t *testing.T, nbc *datamodel.NodeBootstrappingConfiguration) {
+			nbc.ContainerService.Properties.AgentPoolProfiles[0].VMSize = "Standard_D2pds_V5"
+			nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-cblmariner-v2-arm64-gen2"
+			nbc.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.CustomKubeBinaryURL = "https://acs-mirror.azureedge.net/kubernetes/v1.24.9/binaries/kubernetes-node-linux-arm64.tar.gz"
+			nbc.K8sComponents.PodInfraContainerImageURL = "mcr.microsoft.com/oss/kubernetes/pause:3.6"
+			nbc.KubeletConfig["--pod-infra-container-image"] = "mcr.microsoft.com/oss/kubernetes/pause:3.6"
```

Comment:
is this necessary? can we set it in the base tempalte?

Reply:
yeah, probably should anyways since I think 3.5 was recently removed from the cache set anyways - will update

Reply:
it's okay if these aren't cached as long as they work for the most part. sticking closer to the cached/latest bits is useful for latency testing. further drift is useful for "compatibility" testing, looking at it optimistically :D